packages/core/src/server.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { Context, Binding, Constructor } from '@loopback/context';
+
+/**
+ * Defines the requirements to implement a Server for LoopBack applications:
+ * start() : Promise<void>
+ * stop() : Promise<void>
+ * Additionally, an implementation must create a child context instance from
+ * the application
```

Comment:
>I think we need more documentation for people writing custom server implementations - for example, it's important to create another child context for each incoming request too.

Definitely a separate PR, though. :P

Reply:
Agreed. Could you please create a follow-up task to make sure we don't forget? I think it should be labelled as `LoopBackNext-Beta` (because it's for extension developers), probably as `p2` (because I think writing a custom transports is a less common extension)?